pints/_nested/__init__.py

```diff
@@ -50,12 +230,424 @@ def __init__(self, log_likelihood, log_prior):
+        """
+        if threshold <= 0:
+            raise ValueError('Convergence threshold must be positive.')
+        self._marginal_log_likelihood_threshold = threshold
+
+    def marginal_log_likelihood_threshold(self):
+        """
+        Returns threshold for determining convergence in estimate of marginal
+        log likelihood which leads to early termination of the algorithm.
+        """
+        return self._marginal_log_likelihood_threshold
+
+    def iterations(self):
+        """
+        Returns the total number of iterations that will be performed in the
+        next run.
+        """
+        return self._iterations
+
+    def n_posterior_samples(self):
+        """
+        Returns the number of posterior samples that will be returned (see
+        :meth:`set_n_posterior_samples()`).
+        """
+        return self._posterior_samples
+
+    def _initialise_parallel(self, f):
+        """
+        Initialises parallel runners.
+        """
+        # Create evaluator object
+        if self._parallel:
+            # Use at most n_workers workers
+            n_workers = self._n_workers
+            evaluator = pints.ParallelEvaluator(
+                f, n_workers=n_workers)
+        else:
+            evaluator = pints.SequentialEvaluator(f)
+        return evaluator
+
+    def _initialise_sensitivities(self):
+        """
+        Initialises sensitivities if they are needed.
+        """
+        f = self._log_likelihood
+        if self._needs_sensitivities:
+            f = f.evaluateS1
+        return f
+
+    def _initialise_logger(self):
+        """
+        Initialises logger.
+        """
+        # Start logging
+        self._logging = self._log_to_screen or self._log_filename
+        if self._logging:
+            # Create timer
+            self._timer = pints.Timer()
+
+            if self._log_to_screen:
+                # Show current settings
+                print('Running ' + self._sampler.name())
+                print('Number of active points: ' +
+                      str(self._n_active_points))
+                print('Total number of iterations: ' + str(self._iterations))
+                print('Total number of posterior samples: ' + str(
+                    self._posterior_samples))
+
+            # Set up logger
+            self._logger = pints.Logger()
+            if not self._log_to_screen:
+                self._logger.set_stream(None)
+            if self._log_filename:
+                self._logger.set_filename(self._log_filename,
+                                          csv=self._log_csv)
+
+            # Add fields to log
+            self._logger.add_counter('Iter.', max_value=self._iterations)
+            self._logger.add_counter('Eval.', max_value=self._iterations * 10)
+            #TODO: Add other informative fields ?
+            self._logger.add_time('Time m:s')
+            self._logger.add_float('Delta_log(z)')
+            self._logger.add_float('Acceptance rate')
```

Comment:
Maybe this can just create a logger and return it, instead of using `self._logger` ?

Reply:
I had this previously and ended up having to pass the logger to loads of things explicitly, so think having it as a member variables cleans things up.